File: backend/app/interfaces/queue_provider.py
"""Queue Provider Protocol for job processing."""
from typing import Protocol
from app.queue.models import Job


class QueueProvider(Protocol):
    """Protocol for queue implementations (in-process, Azure Service Bus, etc.)."""

    async def enqueue(self, job: Job) -> None:
        """
        Add a job to the queue.

        Args:
            job: Job to enqueue for processing
        """
        ...

    async def start(self) -> None:
        """Start the queue consumer/worker."""
        ...

    async def stop(self) -> None:
        """Stop the queue consumer/worker gracefully."""
        ...

    def is_healthy(self) -> bool:
        """
        Check if queue is operational.

        Returns:
            True if queue is healthy, False otherwise
        """
        ...

File: backend/src/interfaces/queue-provider.ts
import { Job } from '../queue/models.js';

/**
 * Queue Provider interface for job processing.
 */
export interface QueueProvider {
  /** Add a job to the queue. */
  enqueue(job: Job): Promise<void>;
  /** Start the queue consumer/worker. */
  start(): Promise<void>;
  /** Stop the queue consumer/worker gracefully. */
  stop(): Promise<void>;
  /** Check if queue is operational. */
  isHealthy(): boolean;
}
